src/lamp_py/performance_manager/alerts.py

```diff
@@ -0,0 +1,478 @@
+        timestamp_key = f"{key}_timestamp"
+        alerts[key] = (
+            pandas.to_datetime(alerts[timestamp_key], unit="s", utc=True)
+            .dt.tz_convert("America/New_York")
+            .dt.floor("s")
+        )
+
+    return alerts
+
+
+def explode_alerts(alerts: pandas.DataFrame) -> pandas.DataFrame:
+    """
+    the 'informed_entity' column is a list of dicts describing stops
+    along routes in directions that are effected by the alert. explode each
+    record along this list and extract the required information into new
+    columns
+    """
+    alerts = alerts.explode("informed_entity")
+
+    informed_entity_keys = [
+        "route_id",
+        "route_type",
+        "direction_id",
+        "stop_id",
+        "facility_id",
+        "activities",
+    ]
+
+    # extract information from the informed entity
+    for key in informed_entity_keys:
+        alerts[f"informed_entity.{key}"] = alerts["informed_entity"].apply(
+            lambda x, k=key: x.get(k)
+        )
+
+    alerts = alerts.drop(columns=["informed_entity"])
+
+    # transform the activities field from a list to a pipe delimitated string
+    alerts["informed_entity.activities"] = alerts[
+        "informed_entity.activities"
+    ].apply(lambda x: "|".join(str(item) for item in x if item is not None))
+
+    # the commuter rail informed entity contains extra details that aren't
+    # extracted in this transformation. those instances will appear as
+    # duplicated records, so remove them.
+    alerts = alerts.drop_duplicates()
+
+    return alerts
+
+
+def get_unprocessed_alert_files(
+    md_db_manager: DatabaseManager,
+) -> List[Dict[str, str]]:
+    """
+    Get unprocessed RT Alert Files from the MetadataLog table.
+
+    @return [
+        { 'pk_id': 1, 'path': 'lamp/RT_ALERTS/year=2024/month=1/day=4/hour=0/uuid_1-0.parquet' },
+        { 'pk_id': 2, 'path': 'lamp/RT_ALERTS/year=2024/month=1/day=4/hour=1/uuid_2-0.parquet' },
+        ...
+    ]
+    """
+    read_md = sa.select(MetadataLog.pk_id, MetadataLog.path).where(
+        (MetadataLog.rail_pm_processed == sa.false())
+        & (MetadataLog.path.contains("RT_ALERTS"))
+    )
+
+    return md_db_manager.select_as_list(read_md)
+
+
+def process_alerts(md_db_manager: DatabaseManager) -> None:
+    """
+    Gather unprocessed alerts files from springboard, process them, and add them to the public export
+    """
+    process_logger = ProcessLogger("process_alerts")
+    process_logger.log_start()
+    unprocessed_records = get_unprocessed_alert_files(md_db_manager)
+
+    # if there are no new alerts files, exit
+    if len(unprocessed_records) == 0:
+        process_logger.log_complete()
+        return
+
+    # create a handler object that will download, append, and upload data
+    parquet_handler = AlertParquetHandler()
+    existing_id_timestamp_pairs = parquet_handler.existing_id_timestamp_pairs()
+
+    # process up to a month at a time
+    chunk_size = 24 * 30
```

Comment:
Were you able to run this locally with this chunk size? 

During my attempt memory usage spiked over 50GB and I had to kill the process.

Reply:
i dropped it down to a day. i realized in my local testing that it was only testing half a month.